hpat/hiframes/pd_series_ext.py

```diff
@@ -999,13 +999,20 @@ def generic_expand_cumulative_series(self, args, kws):
 if not hpat.config.config_pipeline_hpat_default:
     _not_series_array_attrs.append('resolve_std')
 
+_non_hpat_pipeline_attrs = ['resolve_nsmallest', 'resolve_nlargest']
+
 # use ArrayAttribute for attributes not defined in SeriesAttribute
 for attr, func in numba.typing.arraydecl.ArrayAttribute.__dict__.items():
     if (attr.startswith('resolve_')
             and attr not in SeriesAttribute.__dict__
             and attr not in _not_series_array_attrs):
         setattr(SeriesAttribute, attr, func)
 
+# remove some attributes from SeriesAttribute for non-hpat pipeline
+if not hpat.config.config_pipeline_hpat_default:
+    for attr in _non_hpat_pipeline_attrs:
+        if attr in SeriesAttribute.__dict__:
+            delattr(SeriesAttribute, attr)
```

Comment:
I still don't understand why we need to remove attributes after adding them. I still think it would be better to merge `_non_hpat_pipeline_attrs` with `_not_series_array_attrs` and remove this piece of code.

Reply:
We cannot merge them. The key idea of the code is to remove predefined series attributes from `SeriesAttribute`, not added to `SeriesAttribute` via previous loop. E.g. if to add `'resolve_nsmallest'` to `_not_series_array_attrs` and remove the "deleter" then `SeriesAttribute` will still contain attribute `'resolve_nsmallest'` and "new style" won't be used.